rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,300 @@
+            else:
+                balancer_event_type = BalancerV1EventTypes.EXIT
+                from_event_type = HistoryEventType.RECEIVE
+                ds_address = target_log.address
+
+            action_item = ActionItem(
+                action='transform',
+                sequence_index=target_log.log_index,
+                from_event_type=from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=self.eth,
+                amount=ZERO,
+                to_event_type=None,
+                to_event_subtype=None,
+                to_counterparty=CPT_BALANCER_V1,
+                to_notes=None,
+                extra_data={
+                    'ds_address': ds_address,
+                    'token_address': token_address,
+                    'amount': amount,
+                    'type': balancer_event_type,
+                },
+            )
+            action_items.append(action_item)
+
+        return action_items
+
+    def decode_swap_creation(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: list[ActionItem],
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        """
+        Decode swap in balancer V2. At the beggining of the transaction a SWAP event is created
+        with the information of the tokens and amounts and later some transfers are executed.
+
+        We need to detect this swap event and then match the transferred amounts with the ones
+        in the swap event. A sepcial case is the swap of ETH that is wrapped before being sent.
+        In this case the token is WETH but we have a tranfer of ETH from the user.
+        """
+        if tx_log.topics[0] != V2_SWAP:
+            return None, []
+
+        # The transfer event appears after the swap event, so we need to propagate information
+        from_token = hex_or_bytes_to_address(tx_log.topics[2])
+        to_token = hex_or_bytes_to_address(tx_log.topics[3])
+        amount_in = hex_or_bytes_to_int(tx_log.data[0:32])
+        amount_out = hex_or_bytes_to_int(tx_log.data[32:64])
+        # Create action item to propagate the information about the swap to the transfer enrichers
+        action_item = ActionItem(
+            action='transform',
+            sequence_index=tx_log.log_index,
+            from_event_type=HistoryEventType.RECEIVE,
+            from_event_subtype=HistoryEventSubType.NONE,
+            asset=self.eth,
+            amount=ZERO,
+            to_event_type=None,
+            to_event_subtype=None,
+            to_counterparty=CPT_BALANCER_V2,
+            to_notes=None,
+            extra_data={
+                'from_token': from_token,
+                'to_token': to_token,
+                'amount_in': amount_in,
```

Comment:
you can use asset and amount which are already part of the `ActionItem` for the _in and then extra_data for the _out, but properly document it. Otherwise you end up wasting memory by not using the ActionItem's field and just using the extra data.

Perhaps if this is done in many decoders we may need to rethink how some ActionItems work

If it becomes too complicated you can leave it like this but ... hm ... damn it's ugly to have so many unused fields and use extra data

Reply:
no answer here or resolution by you but I see you use what I suggested, so will resolve this.